added add_route_callback

ayberkenis opened the pull request:
changes made;

- added add_route_callback for easier usage in cogs

### usage example
```py
    async def callback(self, message: str):
        return {}

    async def cog_load(self) -> None:
        await self.ipc.start()
        await self.ipc.add_route_callback(self.callback, 'example_ipc_route')
```

- solved #6 


- ### also added errors module to the `__init__` to be able to use it in quart as it can be imported rn:

```py
@app.errorhandler(winerp.UnauthorizedError)
async def winerp_redirect(error):
    return redirect(url_for("app.index"))
```

AwesomeSam9523 reviewed (approved):
LGTM. Not tested as PR is made by trusted author.

AwesomeSam9523 commented:
@ayberkenis ready for merge?

ayberkenis commented:
new commit has been pushed 4d20d0fe0cea19bb359cf1dee6a13dfa825ecf99

- [x] updated method name from `add_route_callback` to `add_route`

- [x] added docstring for methods that are not protected

- [x] events now has their own class

- [ ] add docs for object transfer

AwesomeSam9523 commented on winerp/__init__.py:
```diff
@@ -9,5 +9,6 @@
 
 from .client import Client
-from .server import Server
+from .lib.errors import *
```
wildcard imports? lol

ayberkenis replied:
oh.. forgot to change it...

AwesomeSam9523 reviewed (approved):
Good job on the Events part.
